### PR TITLE
Feature/l3 195

### PR DESCRIPTION
**Pull request by n3op2**
## What
updates to the `/accept` POST controller to handle rematch2'es

### Changes:
- update to the accept post controller to take `replaces` property and handle rematch2
- eventProcessor.ts update
- off/on chain tests

### CODE REVIEW

File: src/controllers/v1/match2/index.ts [hunk 1]
```diff
@@ -312,16 +288,47 @@ export class Match2Controller extends Controller {
...
+        state: 'submitted',
+        hash: extrinsic.hash.toHex(),
+      })
+
+      this.node.submitRunProcess(extrinsic, this.db.updateTransactionState(transaction.id))
+      return transaction
+    }
+
+    const [oldMatch2]: Match2Row[] = match2.replaces ? await this.db.getMatch2(match2.replaces) : []
+    if (match2.replaces) return acceptRematch()
```
Comment (mattdean-digicatapult) [inline src/controllers/v1/match2/index.ts:320]: the control flow here is pretty ugly. Would it not be better to update the switch statement on ln 331 to return either `acceptFinal` or `acceptRematch` depending on whether this is a rematch or not. I just found it quite hard to reason about what the different checks are doing

File: test/integration/onchain/match2.test.ts [hunk 2]
```diff
@@ -197,6 +197,53 @@ describe('on-chain', function () {
       expect(rematch2.latestTokenId).to.equal(lastTokenId + 4)
     })
 
+    it.only('accepts a rematch2 proposal', async () => {
```
Comment (mattdean-digicatapult) [inline test/integration/onchain/match2.test.ts:200]: only

File: src/controllers/v1/match2/index.ts [hunk 3]
```diff
@@ -312,16 +288,47 @@ export class Match2Controller extends Controller {
...
+    if (state !== 'proposed' && state !== 'acceptedA' && state !== 'acceptedB')
+      throw new BadRequest(`state should not be ${state}`)
```
[the remark this answers is not in the document]
Comment (n3op2) [reply src/controllers/v1/match2/index.ts:322]: this has been removed, it was just a leftover from other tests. The check happens where it used to, been updated.

File: src/controllers/v1/match2/index.ts (hunk 1 above)
Comment (n3op2) [reply src/controllers/v1/match2/index.ts:320]: yes, that has been updated.

File: test/integration/offchain/match2.test.ts [hunk 4]
```diff
@@ -252,6 +256,32 @@ describe('match2', () => {
         updatedAt: exampleDate,
       })
     })
+
+    it.skip('poposes a rematch2', () => {})
```
[the remark this answers is not in the document]
Comment (n3op2) [reply test/integration/offchain/match2.test.ts:260]: was not populated, happy to remove

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**